### PR TITLE
UIOR-626 Clone PO does not clone Prefix and Suffix

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
https://issues.folio.org/browse/UIOR-626

## Approach
`poNumper` consists of `prefix`+`poNumber`+`suffix`
For cloning order, it requires to fetch the next poNumber from BE and then concatenate it with existing prefix and suffix

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/Utils/orderResource.js [hunk 1]
```diff
@@ -44,11 +44,18 @@ export const createOrEditOrderResource = (orderFormValues, mutator) => {
...
 };
 
-export const cloneOrder = (order, mutator, lines) => {
-  const clonedOrder = omit(
-    order,
-    ['id', 'adjustment', 'metadata', 'poNumber', 'poNumberPrefix', 'poNumberSuffix', 'workflowStatus', 'compositePoLines'],
-  );
+export const cloneOrder = async (order, mutator, orderNumberMutator, lines) => {
+  const orderNumberResponse = await orderNumberMutator.GET();
+  const poNumber = orderNumberResponse?.poNumber;
```
Comment (axelhunn) [inline src/components/Utils/orderResource.js:49]: maybe throw here if `poNumber` is undefined?

**Review by aliaksei-chumakou — CHANGES_REQUESTED** [review]
There are separate fields for suffix and prefix, AFAIK https://github.com/folio-org/acq-models/blob/master/mod-orders-storage/schemas/purchase_order.json#L65

File: src/components/Utils/orderResource.js [hunk 2]
```diff
@@ -57,5 +64,5 @@ export const cloneOrder = (order, mutator, lines) => {
     }));
   }
 
-  return saveOrder(clonedOrder, mutator);
+  return poNumber ? saveOrder(clonedOrder, mutator) : Promise.reject();
```
Comment (aliaksei-chumakou) [inline src/components/Utils/orderResource.js:67]: this fn is written using asyn\await, IMO its better to use `throw` instead of `Promise`

Comment (aliaksei-chumakou) [reply src/components/Utils/orderResource.js:67]: also throwing might be done earlier, near the place you've got poNumber

**Comment by AndreiShumski2** [comment]
> There are separate fields for suffix and prefix, AFAIK https://github.com/folio-org/acq-models/blob/master/mod-orders-storage/schemas/purchase_order.json#L65

[here](https://issues.folio.org/browse/UIOR-495) about creating and storing `poNumber` with suffixes and prefixes

**Comment by aliaksei-chumakou** [comment]
IMHO, if number isnt provided - it's populated on the back-end and this should be a back-end fix to use prefix and suffix

**Review by aliaksei-chumakou — APPROVED** [review]

**Review by axelhunn — APPROVED** [review]

**Pull request merged**